server/athenian/api/experiments/preloading/entries.py

```diff
@@ -116,6 +119,148 @@ class PreloadedReleaseToPullRequestMapper(ReleaseToPullRequestMapper):
+        """
+        Load some data belonging to released or rejected PRs from the preloaded precomputed DB.
+
+        Query version. JIRA must be filtered separately.
+        :return: 1. Map PR node ID -> repository name & specified column value. \
+                 2. Map from repository name to ambiguous PR node IDs which are released by \
+                 branch with tag_or_branch strategy and without tags on the time interval.
+        """
+        if labels:
+            # TODO: not supported yet, call original implementation
+            return await super(PreloadedDonePRFactsLoader, cls)\
+                ._load_precomputed_done_filters(
+                    columns, time_from, time_to, repos, participants, labels,
+                    default_branches, exclude_inactive, release_settings, account, pdb)
+
+        cached_df = pdb.cache.dfs[PCID.done_pr_facts]
+        df = cached_df.df
+
+        with sentry_sdk.start_span(op="_load_precomputed_done_filters/mask_creation"):
+            mask = df["acc_id"] == account
+            if time_from is not None:
+                mask &= df["pr_done_at"] >= time_from
+            if time_to is not None:
+                mask &= df["pr_created_at"] < time_to
+            if repos is not None:
+                mask &= df["repository_full_name"].isin(repos)
+
+            if len(participants) > 0:
+                mask &= cls._build_participants_mask(df, participants)
+
+            if exclude_inactive:
+                with sentry_sdk.start_span(
+                    op="_load_precomputed_done_filters/mask_creation/activity_days"
+                ):
+                    activity_days = np.concatenate(df["activity_days"].values)
+                    activity_mask = np.full(len(df["activity_days"]), False)
+                    activity_days_in_range = (
+                        time_from.replace(tzinfo=timezone.utc) <= activity_days
+                    ) & (activity_days < time_to.replace(tzinfo=timezone.utc))
+                    activity_offsets = np.cumsum(df["activity_days"].apply(len).values)
+                    indexes = np.searchsorted(
+                        activity_offsets,
+                        np.nonzero(activity_days_in_range)[0],
+                        side="right",
+                    )
+                    activity_mask[indexes] = 1
+
+                    mask &= activity_mask
+
+            match_groups, event_repos, _ = group_repos_by_release_match(
+                repos, default_branches, release_settings)
+            match_groups[ReleaseMatch.rejected] = match_groups[
+                ReleaseMatch.force_push_drop
+            ] = {"": repos}
+            if event_repos:
+                match_groups[ReleaseMatch.event] = {"": event_repos}
+
+            mask &= cls._match_groups_to_mask(df, match_groups)
+
+        done_pr_facts = cached_df.filter(
+            mask, columns=["pr_node_id", "repository_full_name", "release_match"] + [
```

Comment:
```suggestion
            mask, columns=[DonePullRequestFacts.pr_node_id.key, "repository_full_name", "release_match"] + [
```
There should be zero hardcoded strings like these in the code because if need a refactoring with a column you can click "Find Usages" in PyCharm and simply go through the list.

Reply:
I'll do a separate PR before this and rebase where I fix all the names only.